File: 704-binary-search/704-binary-search.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        return binarySearch(nums, target);
    }
private:
    int binarySearch(vector<int>& nums, int target) {
        int low = 0, high = nums.size() - 1;
        
        while(low <= high) {
            int mid = low + (high - low) / 2;
            
            if(nums[mid] == target)
                return mid;
            else if(target < nums[mid])
                high = mid - 1;
            else
                low = mid + 1;
        }
        return -1;
    }
};

File: 704-binary-search/704-binary-search.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        beg = 0
        end = len(nums) - 1
        
        while beg <= end:
            
            mid = (beg + end) // 2
            
            if nums[mid] == target:
                return mid
            
            elif target < nums[mid]:
                end = mid - 1
            else:
                beg = mid + 1
        return -1